src/operators/reduce-support.ts

```diff
@@ -0,0 +1,57 @@
+import Operator from '../Operator';
+import Observer from '../Observer';
+import Subscriber from '../Subscriber';
+
+import tryCatch from '../util/tryCatch';
+import {errorObject} from '../util/errorObject';
+
+export class ReduceOperator<T, R> implements Operator<T, R> {
+
+  acc: R;
+  project: (acc: R, x: T) => R;
+
+  constructor(project: (acc: R, x: T) => R, acc?: R) {
```

Comment:
@kwonoj I think that it's better leave this change to a new pull request. I didn't changed the code, just moved to a new file. There are some other minor changes such as a better signature, `seed` instead of `acc`. What do you think?